src/sergittos/bedwars/listener/GameListener.php

```diff
@@ -403,6 +403,10 @@ public function onPickup(EntityItemPickupEvent $event): void {
             foreach($entity->getInventory()->addItem($event->getItem()) as $remains) {
                 $world->dropItem($origin->getLocation(), $remains, new Vector3(0, 0, 0));
             }
+
+            if ($entity->getGamemode() === GameMode::SPECTATOR){
```

Comment:
GameMode uses dynamic registry of functions, please change to `GameMode::SPECTATOR()`

Reply:
That would be fine, even so i didn't tested out.